feat: template pull requests title and body

alexymantha opened the pull request:
Allows for templating the pull request title and body.

I added the templates to the GitRepository CRD because I feel like it makes sense to allow each repository to configure how they want the pull requests to be titled. I think this allows the most flexibility. I'm open to suggestion if you think that should live elsewhere.

Here is an example of a PR that was created using a custom title template: https://github.com/alexymantha/gitops-test/pull/3/files. It was created with the following configuration:
```
kind: GitRepository
metadata:
  name: gitops-test
  namespace: promoter-system
spec:
  name: gitops-test
  owner: alexymantha
  scmProviderRef:
    name: gitops-test
  titleTemplate: 'chore: promote {{ trunc 5 .Status.Proposed.Dry.Sha }} to `{{ .Spec.ActiveBranch
    }}`'

```

dependabot[bot] committed:
chore(deps): bump sigs.k8s.io/controller-runtime from 0.20.2 to 0.20.3 (#209)

Bumps [sigs.k8s.io/controller-runtime](https://github.com/kubernetes-sigs/controller-runtime) from 0.20.2 to 0.20.3.
- [Release notes](https://github.com/kubernetes-sigs/controller-runtime/releases)
- [Changelog](https://github.com/kubernetes-sigs/controller-runtime/blob/main/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/controller-runtime/compare/v0.20.2...v0.20.3)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/controller-runtime
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

crenshaw-dev committed:
chore: mark generated files as generated to suppress diffs in PRs (#210)

Signed-off-by: Michael Crenshaw <350466+crenshaw-dev@users.noreply.github.com>

Oded-B committed:
docs: Update Telefonistka project URL to reflect its new home (#212)

Old repo was archived and a comment was placed in the top of the README.md if verification is needed.

Signed-off-by: Oded Ben Ozer <oded.benozer@gmail.com>

dependabot[bot] committed:
chore(deps): bump the k8s-deps group with 3 updates (#211)

Bumps the k8s-deps group with 3 updates: [k8s.io/api](https://github.com/kubernetes/api), [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery) and [k8s.io/client-go](https://github.com/kubernetes/client-go).


Updates `k8s.io/api` from 0.32.2 to 0.32.3
- [Commits](https://github.com/kubernetes/api/compare/v0.32.2...v0.32.3)

Updates `k8s.io/apimachinery` from 0.32.2 to 0.32.3
- [Commits](https://github.com/kubernetes/apimachinery/compare/v0.32.2...v0.32.3)

Updates `k8s.io/client-go` from 0.32.2 to 0.32.3
- [Changelog](https://github.com/kubernetes/client-go/blob/master/CHANGELOG.md)
- [Commits](https://github.com/kubernetes/client-go/compare/v0.32.2...v0.32.3)

---
updated-dependencies:
- dependency-name: k8s.io/api
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: k8s-deps
- dependency-name: k8s.io/apimachinery
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: k8s-deps
- dependency-name: k8s.io/client-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: k8s-deps
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump docker/login-action from 3.3.0 to 3.4.0 (#213)

Bumps [docker/login-action](https://github.com/docker/login-action) from 3.3.0 to 3.4.0.
- [Release notes](https://github.com/docker/login-action/releases)
- [Commits](https://github.com/docker/login-action/compare/9780b0c442fbb1117ed29e0efdff1e18412f7567...74a5d142397b4f367a81961eba4e8cd7edddf772)

---
updated-dependencies:
- dependency-name: docker/login-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alexymantha committed:
docs: reformat GitHub app documentation (#218)

Signed-off-by: Alexy Mantha <alexy@mantha.dev>

alexymantha committed:
starter implementation

Signed-off-by: Alexy Mantha <alexy@mantha.dev>

dependabot[bot] committed:
chore(deps): bump actions/setup-go from 5.3.0 to 5.4.0 (#219)

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 5.3.0 to 5.4.0.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/f111f3307d8850f501ac008e886eec1fd1932a34...0aaccfd150d50ccaeb58ebd88d36e91967a5f35b)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alexymantha committed:
fix: wrap pull request creation error (#217)

Signed-off-by: Alexy Mantha <alexy@mantha.dev>

crenshaw-dev commented on internal/controller/changetransferpolicy_controller.go:
```diff
@@ -473,3 +486,36 @@ func (r *ChangeTransferPolicyReconciler) mergePullRequests(ctx context.Context,
+	if bodyTemplate == "" {
+		bodyTemplate = DefaultPRBodyTemplate
+	}
+	tmpl, err = template.New("body").Funcs(sprig.FuncMap()).Parse(bodyTemplate)
```
Will probably want to pop `env`, `expandenv`, and `getHostByName` out of the function map, just to be a bit more secure (Argo CD, Events, and Workflows all do this as well).

zachaller committed:
feat: add annotation for on copied active commit status (#216)

* feat: add annotation for copied commit status

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: update if annotation changes

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: update if annotation changes

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: add const

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: rename

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: scope down

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: consistency

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* Update api/v1alpha1/constants.go

Co-authored-by: Michael Crenshaw <350466+crenshaw-dev@users.noreply.github.com>
Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: review

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: rename

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: rename

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

* fix: rename

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>

---------

Signed-off-by: Zach Aller <zachaller@users.noreply.github.com>
Co-authored-by: Michael Crenshaw <350466+crenshaw-dev@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.35.0 to 0.36.0 (#220)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.35.0 to 0.36.0.
- [Commits](https://github.com/golang/net/compare/v0.35.0...v0.36.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

alexymantha committed:
remove unwanted sprig functions

Signed-off-by: Alexy Mantha <alexy@mantha.dev>

alexymantha committed:
fix linting

Signed-off-by: Alexy Mantha <alexy@mantha.dev>

alexymantha committed:
codegen

Signed-off-by: Alexy Mantha <alexy@mantha.dev>

alexymantha committed:
gofumpt

Signed-off-by: Alexy Mantha <alexy@mantha.dev>

alexymantha committed:
Merge branch 'main' into feat/template-pr-title-and-body

alexymantha commented:
https://github.com/argoproj-labs/gitops-promoter/pull/224
